api/src/main/java/org/pmiops/workbench/elasticsearch/ElasticSearchService.java

```diff
@@ -0,0 +1,55 @@
+package org.pmiops.workbench.elasticsearch;
+
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.client.RestHighLevelClient;
+import org.elasticsearch.client.core.CountRequest;
+import org.elasticsearch.index.query.AbstractQueryBuilder;
+import org.elasticsearch.index.query.BoolQueryBuilder;
+import org.elasticsearch.index.query.TermQueryBuilder;
+import org.elasticsearch.search.builder.SearchSourceBuilder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+
+@Service
+public class ElasticSearchService {
+
+  private static final String INDEX = "index";
+  private static final String TYPE = "type";
+  private RestHighLevelClient client;
+
+  @Autowired
+  public ElasticSearchService(RestHighLevelClient client) {
+    this.client = client;
+  }
+
+  /**
+   * This method will be used to generate counts from a
+   * org.pmiops.workbench.model.SearchRequest.
+   *
+   * @param userRequest
+   * @return
+   */
+  public Long elasticCount(org.pmiops.workbench.model.SearchRequest userRequest) throws IOException {
+    CountRequest countRequest = new CountRequest()
+      //What indices should we use here?
+      .indices(INDEX, TYPE)
+      .source(new SearchSourceBuilder().query(createQueryBuilder(userRequest)));
+    return client.count(countRequest, RequestOptions.DEFAULT).getCount();
+  }
+
+  //Is it reasonable to user QueryBuilders for this? Or do we need something more flexible to
```

Comment:
Good question, we should think about what interfaces we want here. My high level idea was that there would somehow be a common interface for use between BigQuery and Elastic to get counts. I'm not sure whether this holds up though.

Reply:
Ok